adapters/applogy/applogy.go

```diff
@@ -0,0 +1,157 @@
+package applogy
+
+import (
+	"encoding/json"
+	"errors"
+	"net/http"
+	"strconv"
+
+	"github.com/mxmCherry/openrtb"
+	"github.com/prebid/prebid-server/adapters"
+	"github.com/prebid/prebid-server/errortypes"
+	"github.com/prebid/prebid-server/openrtb_ext"
+)
+
+type ApplogyAdapter struct {
+	endpoint string
+}
+
+func (a *ApplogyAdapter) MakeRequests(request *openrtb.BidRequest, _ *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
+	headers := http.Header{}
+	headers.Add("Content-Type", "application/json;charset=utf-8")
+	headers.Add("Accept", "application/json")
+	impressions := request.Imp
+	totalImpressions := len(impressions)
+	result := make([]*adapters.RequestData, 0, totalImpressions)
```

Comment:
no need to declare `totalImpressions` since `len` reads from memory rather than counting items and, in consequence, comes at a time complexity of O(1)